### PR TITLE
port to modern world

### PR DESCRIPTION
**Pull request by devkral**
Changes:
- upgrade postgres to use the psycopg 3 driver
- skip mssql when not odbc driver installed
- remove aiopg (no real update for 2 years) and better alternatives available
- use hatch environments
- use hatch test and fmt commands
- use mariadb instead of mysql (fixes memory exhaustion)
- <del>use iterate for fetch_all (chunking would be nice)</del>

TODO:
- <del>remove cruft</del>
- <del>update ga actions</del>
- <del>investigate appearing testsuite file</del>
- <del>fix/replace scripts</del>

### CODE REVIEW

File: tests/test_connection_options.py [hunk 1]
```diff
@@ -30,25 +30,25 @@ async def test_postgres_pool_size_connect():
 
 
 def test_postgres_explicit_pool_size():
-    backend = PostgresBackend("postgres://localhost/database", min_size=1, max_size=20)
+    backend = PostgresBackend("postgres+psycopg://localhost/database", min_size=1, max_size=20)
```
Comment (tarsil) [inline tests/test_connection_options.py:33]: this was correct because it was testing for the driver without async aschema as well

**Review by tarsil — COMMENTED** [review]
The same for the rest

File: tests/test_connection_options.py (hunk 1 above)
Comment (devkral) [reply tests/test_connection_options.py:33]: psycopg handles both ;)

It is simply an upgrade to psycopg3

Comment (tarsil) [reply tests/test_connection_options.py:33]: Odd, I was sure I had that, anyway good catch. We have the CI failing but will be good to add a develop branch from the main and open against that?

**Comment by devkral** [comment]
nice, the failing tests were gone

**Comment by devkral** [comment]
I think I have an ok. Let's merge.

**Pull request merged**